api/views.py

```diff
@@ -35,3 +36,46 @@ def get_queryset(self):
             kwargs["date__lte"] = end_date
 
         return self.queryset.filter(**kwargs)
+
+
+class CityHallBidView(ListAPIView):
+    serializer_class = CityHallBidSerializer
+
+    def get_queryset(self):
+        bids = CityHallBid.objects.prefetch_related("events").prefetch_related("files")
+
+        if self.request.query_params:
+            bids = self.filter_by_query_params(bids)
+
+        return bids
+
+    def filter_by_query_params(self, bids):
+        bids = self.filter_by_query(bids)
+        bids = self.filter_by_start_date(bids)
+        bids = self.filter_by_end_date(bids)
+        return bids
+
+    def filter_by_end_date(self, bids):
+        end_date = self.request.query_params.get("end_date", None)
+        if end_date is not None:
+            bids = bids.filter(session_at__date__lte=end_date)
+        return bids
+
+    def filter_by_start_date(self, bids):
+        start_date = self.request.query_params.get("start_date", None)
+        if start_date is not None:
+            bids = bids.filter(session_at__date__gte=start_date)
+        return bids
+
+    def filter_by_query(self, bids):
+        description = self.request.query_params.get("query", None)
+
+        if description:
+            description = description.replace('"', "")
+            search_vector = SearchVector("files__search_vector", config="portuguese")
+            bids = (
+                bids.filter(description__icontains=description)
+                | bids.filter(events__summary__icontains=description)
+                | bids.annotate(search=search_vector).filter(search=description)
```

Comment:
@joorgelm como está a performance por aí? Aqui está bem lenta quando buscando pela `query`. Aqui o termo que estou buscando:
localhost:8000/api/city-hall/bids/?query=corel
http://localhost:8000/api/city-hall/bids/?query=tomba
Rodei todos os crawlers e o banco tem cerca de 8 mil arquivos com conteúdo. Meu computador tem 16GB de RAM e roda num i7. O endpoint está levando mais de 40 segundos pra responder - o que não dá pra gente aceitar pq nossos servidores vão rodar numa configuração bem mais modesta.
Me avise se precisar de um dump do banco.

Reply:
Você pode usar o [explain](https://docs.djangoproject.com/en/3.1/topics/db/optimization/#use-explain) para debugar isso melhor.

Reply:
a performance está bem parecida com a sua, estou fazendo alguns testes p tentar encontrar o maior gargalo.

se não for problemático, gostaria do dump do banco sim :)

Obrigado.

Reply:
Me manda msg no privado no nosso servidor do discord que eu te passo. :) @joorgelm